src/server/auth.ts

```diff
@@ -74,3 +75,50 @@ export const getServerAuthSession = (ctx: {
 }) => {
   return getServerSession(ctx.req, ctx.res, authOptions);
 };
+
+
+/*
+todo write docs
+*/
+export interface session {
+  account : Number;
+  token : String;
+  expiry : Date;
+}
+
+export const validateUser = async (
+  id:number, 
+  token:string
+  ) => {
+
+    const result = await prisma.session
+    .findMany({where: 
+            { 
+              account: id,
+              token : token 
+            } } )
+    .catch(message => {
+              throw new TRPCError({
+                 code: "INTERNAL_SERVER_ERROR",
+                 message,
+              })})
+    
+   const now = new Date();
+
+    if(!result[0])
+    {
+      throw new Error("Session does not currently exist");
+    }
+    else if(result[0]?.expiry < now)
+    {
+      throw new Error("Session has expired");
```

Comment:
```suggestion
      throw new Error(`Session expired at ${sessionExpireTime.toLocaleDateString()}`);
```

I try to be extra verbose when it comes to error messages.